### PR TITLE
Functional test

### PR DESCRIPTION
**Pull request by LuitelSamikshya**

### CODE REVIEW

File: cmd/functional-test/testcases.txt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+www.example.com {{binary}} -silent -l test-data/request.txt
+www.example.com {{binary}} -silent -request test-data/raw-request.txt
+www.example.com {{binary}} -silent -response-time
+www.example.com {{binary}} -silent -title
+www.example.com {{binary}} -silent -location
+www.example.com {{binary}} -silent -cname
+www.example.com {{binary}} -silent -cdn
+www.example.com {{binary}} -silent -probe
+www.example.com {{binary}} -silent -no-fallback
+www.example.com {{binary}} -silent -match-code 200, -mc 200
```
Comment (Mzack9999) [inline cmd/functional-test/testcases.txt:11]: I think we should remove the comma after the status code

File: cmd/functional-test/testcases.txt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+www.example.com {{binary}} -silent -match-length 100, -ml 100
```
Comment (Mzack9999) [inline cmd/functional-test/testcases.txt:12]: I think we should remove the comma after the status code

File: cmd/functional-test/testcases.txt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+www.example.com {{binary}} -silent -match-string xyz, -ms xyz
```
Comment (Mzack9999) [inline cmd/functional-test/testcases.txt:13]: I think the comma shouldn't be necessary here

File: cmd/functional-test/testcases.txt [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+www.example.com {{binary}} -silent -match-regex bbb, -mr bbb
```
Comment (Mzack9999) [inline cmd/functional-test/testcases.txt:14]: I think the comma shouldn't be necessary here

File: cmd/functional-test/testcases.txt [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+www.example.com {{binary}} -silent -filter-code 401, -fc 401
```
Comment (Mzack9999) [inline cmd/functional-test/testcases.txt:15]: I think the comma shouldn't be necessary here

File: cmd/functional-test/testcases.txt [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+www.example.com {{binary}} -silent -filter-length 23, -fl 23
```
Comment (Mzack9999) [inline cmd/functional-test/testcases.txt:16]: I think the comma shouldn't be necessary here

File: cmd/functional-test/testcases.txt [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+www.example.com {{binary}} -silent -filter-string
```
Comment (Mzack9999) [inline cmd/functional-test/testcases.txt:17]: missing the strings argument to filter

File: cmd/functional-test/testcases.txt [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+www.example.com {{binary}} -silent -threads 50
+www.example.com {{binary}} -silent -rate-limit 5
+www.example.com {{binary}} -silent -tls-grab
+www.example.com {{binary}} -silent -pipeline
+www.example.com {{binary}} -silent -http2
+www.example.com {{binary}} -silent -ports
```
Comment (Mzack9999) [inline cmd/functional-test/testcases.txt:23]: missing comma separated ports

File: cmd/functional-test/testcases.txt [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+www.example.com {{binary}} -silent -paths
```
Comment (Mzack9999) [inline cmd/functional-test/testcases.txt:24]: missing the paths arguments (either from file or comma separated)

File: cmd/functional-test/testcases.txt [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+www.example.com {{binary}} -silent -response-size-to-read 1028
+www.example.com {{binary}} -silent -allow 127.0.0.1:8080
+www.example.com {{binary}} -silent -deny 127.0.0.1:6089
+www.example.com {{binary}} -random-agent
+www.example.com {{binary}} -silent -proxy www.example.com
+www.example.com {{binary}} -silent -unsafe
+www.example.com {{binary}} -silent -resume
+www.example.com {{binary}} -silent -max-redirects 10
+www.example.com {{binary}} -silent -vhost-input
+www.example.com {{binary}} -silent -x
```
Comment (Mzack9999) [inline cmd/functional-test/testcases.txt:40]: missing method argument value

File: cmd/functional-test/testcases.txt [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+www.example.com {{binary}} -silent -body
```
Comment (Mzack9999) [inline cmd/functional-test/testcases.txt:41]: missing body value

File: cmd/functional-test/testcases.txt (hunk 1 above)
Comment (LuitelSamikshya) [reply cmd/functional-test/testcases.txt]: Done.

File: cmd/functional-test/testcases.txt (hunk 2 above)
Comment (LuitelSamikshya) [reply cmd/functional-test/testcases.txt]: Done.

File: cmd/functional-test/testcases.txt (hunk 3 above)
Comment (LuitelSamikshya) [reply cmd/functional-test/testcases.txt]: Done.

File: cmd/functional-test/testcases.txt (hunk 4 above)
Comment (LuitelSamikshya) [reply cmd/functional-test/testcases.txt]: Done.

File: cmd/functional-test/testcases.txt (hunk 5 above)
Comment (LuitelSamikshya) [reply cmd/functional-test/testcases.txt]: Done.

File: cmd/functional-test/testcases.txt (hunk 6 above)
Comment (LuitelSamikshya) [reply cmd/functional-test/testcases.txt]: Done.

File: cmd/functional-test/testcases.txt (hunk 7 above)
Comment (LuitelSamikshya) [reply cmd/functional-test/testcases.txt]: Done.

File: cmd/functional-test/testcases.txt (hunk 8 above)
Comment (LuitelSamikshya) [reply cmd/functional-test/testcases.txt]: Done.

File: cmd/functional-test/testcases.txt (hunk 9 above)
Comment (LuitelSamikshya) [reply cmd/functional-test/testcases.txt]: Done.

File: cmd/functional-test/testcases.txt (hunk 10 above)
Comment (LuitelSamikshya) [reply cmd/functional-test/testcases.txt]: Done.

File: cmd/functional-test/testcases.txt (hunk 11 above)
Comment (LuitelSamikshya) [reply cmd/functional-test/testcases.txt]: Done.

**Review by Mzack9999 — APPROVED** [review]

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**